core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1490,14 +1494,14 @@ object FiberRuntime {
 
   private[zio] val catastrophicFailure: AtomicBoolean = new AtomicBoolean(false)
 
-  private val fiberFailureTracker: Cause.Folder[Unit, Any, Unit] =
+  private def fiberFailureTracker(tags: Set[MetricLabel]): Cause.Folder[Unit, Any, Unit] =
```

Comment:
This degrades performance. Instead of making this a `def`, passing in the `tags` as part of the context of the folder.